src/main/java/bio/terra/cli/businessobject/User.java

```diff
@@ -119,20 +121,41 @@ public void loadExistingCredentials() {
     }
   }
 
+  /**
+   * Load any existing credneitals for the user. Prompt for login if they are expired or do not
+   * exist. Do not use application default credentials for logging in.
+   */
+  public static void login() {
+    login(false);
+  }
+
   /**
    * Load any existing credentials for this user. Prompt for login if they are expired or do not
    * exist.
+   *
+   * @param useAdc whether to use application default credentials to log in.
    */
-  public static void login() {
+  public static void login(boolean useAdc) {
     Optional<User> currentUser = Context.getUser();
     currentUser.ifPresent(User::loadExistingCredentials);
 
     // populate the current user object or build a new one
     User user = currentUser.orElseGet(() -> new User());
 
-    // do the login flow if the current user is undefined or has expired credentials
-    if (currentUser.isEmpty() || currentUser.get().requiresReauthentication()) {
-      user.doOauthLoginFlow();
+    if (useAdc) {
+      try {
+        user.checkForAppDefaultCredentials();
+      } catch (UserActionableException e) {
+        logger.debug("Failed to get app default credentials, do oauth login flow instead");
+        // do the login flow if the current user is undefined or has expired credentials
+        if (currentUser.isEmpty() || currentUser.get().requiresReauthentication()) {
+          user.doOauthLoginFlow();
```

Comment:
imo I think this should throw the UserActionableException instead of falling back to the login flow. 
In this case the user has explicitly specified the `--use-app-default-credentials` flag, so I think they would expect the CLI to either use ADC or throw, rather than falling back to ignoring the flag.